### ISSUE TITLE
Impeller: Animated Widget placed above ImageFiltered with Blur causes UI Glitches

### ISSUE THREAD

**Issue by b-ray** [flutter/flutter#135013]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Run the sample code provided - with a Stack containing a Blurred Container below an Animated Element (in this case a animating `CupertinoActivityIndicator`)

__________________________

I was able to reproduce this on:
- `3.15.0-15.1.pre` with Impeller
- `3.13.4` with Impeller

The issue does not occur when using Skia.

(I am able to work around the glitches using `BackdropFilter`, but I am not sure if this is proper fix - I am worried about performance issues)

### Expected results

The animation and surrounding box should be drawn properly.

### Actual results

The surrounding box of the animation has glitches. Please see screenshot.

### Code sample

<details><summary>Code sample</summary>

```dart
import 'dart:ui';

import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Stack(
        children: [
          ImageFiltered(
            imageFilter: ImageFilter.blur(sigmaX: 40, sigmaY: 40),
            child: Container(
              decoration: const BoxDecoration(
                gradient: RadialGradient(
                  radius: 0.3,
                  colors: [Colors.pink, Colors.green],
                ),
              ),
            ),
          ),
          const Center(
            child: CupertinoActivityIndicator(
              radius: 30,
              animating: true,
            ),
          )
        ],
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

Screenshots are taken using Flutter 3.13.4.

Skia (Expected Result):
![Simulator Screenshot - iPhone 11 Pro - 2023-09-19 at 12 11 57](https://github.com/flutter/flutter/assets/1121929/cc06b2fa-4ade-4512-b85e-7b08475eef8d)

Impeller (Actual Result):
![Simulator Screenshot - iPhone 11 Pro - 2023-09-19 at 12 11 04](https://github.com/flutter/flutter/assets/1121929/9e6cf8b8-72f0-4a57-a5bb-fe8b54e46fa0)

</details>

### Logs

<details><summary>No Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.13.4, on macOS 13.5.2 22G91 darwin-arm64, locale en-AT)
    • Flutter version 3.13.4 on channel stable at /Users/stpu/fvm/versions/3.13.4
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 367f9ea16b (6 days ago), 2023-09-12 23:27:53 -0500
    • Engine revision 9064459a8b
    • Dart version 3.1.2
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/stpu/Library/Android/sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • ANDROID_HOME = /Users/stpu/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode-14.3.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.1

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.82.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.72.0

[✓] Connected device (2 available)
    • iPhone 11 Pro (mobile) • 8502B9A2-1553-4219-93B0-21E155C971E7 • ios          • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64 • macOS 13.5.2 22G91 darwin-arm64

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by dam-ease** [comment]
Hi @b-ray. Thanks for filing this. I can reproduce this on the latest `stable` channel `3.13.4` but can't seem to reproduce on the latest `master` channel `3.14.0-14.0.pre.358` both with and without impeller. Are you able to check this on your end as well?

**Comment by b-ray** [comment]
Thanks very much @dam-ease , the problem shown by my sample code above seems indeed fixed on the latest `master` version `3.14.0-14.0.pre.358`. That's great!

Unfortunately I am still seeing major glitches in our Production App when using version `3.14.0-14.0.pre.358`.

I've updated my sample code to demonstrate the problem I am seeing with `3.14.0-14.0.pre.358` here:

<details><summary>New Sample Code</summary>
<p>

```dart
import 'dart:ui';

import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Stack(
        children: [
          ImageFiltered(
            imageFilter: ImageFilter.blur(sigmaX: 40, sigmaY: 40),
            child: ColorFiltered(
              colorFilter: ColorFilter.mode(
                  Colors.black.withOpacity(0.2), BlendMode.srcOver),
              child: Container(
                decoration: const BoxDecoration(
                  gradient: RadialGradient(
                    radius: 0.3,
                    colors: [Colors.pink, Colors.green],
                  ),
                ),
              ),
            ),
          ),
          const Center(
            child: CupertinoActivityIndicator(
              radius: 30,
              animating: true,
            ),
          )
        ],
      ),
    );
  }
}

``` 

</p>
</details> 

This code introduced a `ColorFiltered` Widget on top of the `ImageFiltered` Widget, causing further glitches.

<details><summary>Screenshots</summary>
<p>

Skia (Expected Result):
![Simulator Screenshot - iPhone 11 Pro - 2023-09-19 at 17 52 33](https://github.com/flutter/flutter/assets/1121929/3d5a2647-d284-44a2-96c3-dc024f179133)

Impeller (Actual Result):
![Impeller](https://github.com/flutter/flutter/assets/1121929/6db38592-858d-4455-ab67-2792925adf11)


</p>
</details> 

When adding code to move the Activity Indicator up and down, the glitches become much worse.

Any help would be very much appreciated.

**Comment by dam-ease** [comment]
Hi @b-ray. Thanks for this. I can reproduce this new issue you described. Can you please file a new issue describing the issue with adding the `ColorFiltered` widget so it's easier to track? I'd be closing this as a fixed in the latest versions.
Thank you!
